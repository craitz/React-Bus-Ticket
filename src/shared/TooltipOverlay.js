import React from 'react';
import { Tooltip, OverlayTrigger } from 'react-bootstrap';

const TooltipOverlay = ({ text, position = "bottom", children }) => {
  const tooltip = <Tooltip id="tooltip">{text}</Tooltip>
  return (
    <OverlayTrigger placement={position} overlay={tooltip}>
      {children}
    </OverlayTrigger>
  );
};

export default TooltipOverlay;