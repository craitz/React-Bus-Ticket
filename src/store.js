import { applyMiddleware, createStore } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import reducer from './reducers';

const configureStore = () => {
  const logger = createLogger({
    collapsed: true,
    duration: true,
    diff: true,
    level: ''
  });

  const middleware = applyMiddleware(promise(), thunk, logger);

  return createStore(reducer, middleware);
}

export default configureStore();
