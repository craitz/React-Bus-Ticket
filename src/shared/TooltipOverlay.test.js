import React from 'react';
import store from '../store';
import { shallow } from 'enzyme';
import TooltipOverlay from './TooltipOverlay';

describe('TooltipOverlay VIEW', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <TooltipOverlay />
    );
    expect(wrapper).toMatchSnapshot();
  });
});