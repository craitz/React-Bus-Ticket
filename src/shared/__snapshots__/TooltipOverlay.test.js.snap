// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipOverlay VIEW renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
        <Tooltip
            bsClass="tooltip"
            id="tooltip"
            placement="right"
        />
    }
    placement="bottom"
    trigger={
        Array [
            "hover",
            "focus",
          ]
    }
/>,
  "nodes": Array [
    <OverlayTrigger
      defaultOverlayShown={false}
      overlay={
            <Tooltip
                  bsClass="tooltip"
                  id="tooltip"
                  placement="right"
            />
      }
      placement="bottom"
      trigger={
            Array [
                  "hover",
                  "focus",
                ]
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TooltipOverlay
        position="bottom"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "position": "bottom",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
                    <Tooltip
                              bsClass="tooltip"
                              id="tooltip"
                              placement="right"
                    />
          }
          placement="bottom"
          trigger={
                    Array [
                              "hover",
                              "focus",
                            ]
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
                    <Tooltip
                              bsClass="tooltip"
                              id="tooltip"
                              placement="right"
                    />
          }
          placement="bottom"
          trigger={
                    Array [
                              "hover",
                              "focus",
                            ]
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TooltipOverlay
    position="bottom"
/>,
}
`;
